Library/OcCpuLib/OcCpuLib.c

```diff
@@ -1177,14 +1180,16 @@ ScanAmdProcessor (
   //
   Cpu->AppleProcessorType = AppleProcessorTypeCorei5Type5;
   //
-  // get TSC Frequency calculated in OcTimerLib
+  // get TSC Frequency calculated in OcTimerLib, only if we get it already when it under visualization.
   // FIXME(1): This code assumes the CPU operates in P0.  Either ensure it does
   //           and raise the mode on demand, or adapt the logic to consider
   //           both the operating and the nominal frequency, latter for
   //           the invariant TSC.
   //
-  Cpu->CPUFrequencyFromTSC = OcCalculateTSCFromPMTimer (Recalculate);
-  Cpu->CPUFrequency = Cpu->CPUFrequencyFromTSC;
+    if (!(Cpu->Hypervisor && (Cpu->Features & CPUID_EXTFEATURE_TSCI))){
```

Comment:
There is an extra indentation here. This line and the 2 below need to be moved 2 spaces left.